### PR TITLE
Add Thrift support for ResourceGroupInfo

### PR DESCRIPTION
**Pull request by Weizitong**
Add Thrift support for ResourceGroupInfo.
A new TestThriftResourceGroupInfo is added.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by ajaygeorge** [comment]
reviewed all main classes LGTM.
pending test class.

File: presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        assertEquals(actualQueryStateInfoList.size(), 2);
+
+        assertEquals(actualQueryStateInfoList.get(0).getQueryId(), FAKE_QUERY_ID_1);
+        asserSerdeQueryState(actualQueryStateInfoList.get(0), FAKE_PROGRESS_PERCENTAGE_1);
+
+        assertEquals(actualQueryStateInfoList.get(1).getQueryId(), FAKE_QUERY_ID_2);
+        asserSerdeQueryState(actualQueryStateInfoList.get(1), FAKE_PROGRESS_PERCENTAGE_2);
+    }
+
+    private void asserSerdeQueryState(QueryStateInfo actualInfo, OptionalDouble expectedProgress)
```
Comment (swapsmagic) [inline presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java:209]: `assertSerdeQueryState`

**Review by swapsmagic — APPROVED** [review]
minor typo but rest looks good.

File: presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestThriftResourceGroupInfo
+{
+    private static final ThriftCatalog COMMON_CATALOG = new ThriftCatalog();
+    private static final DataSizeToBytesThriftCodec DATA_SIZE_CODEC = new DataSizeToBytesThriftCodec(COMMON_CATALOG);
+    private static final JodaDateTimeToEpochMillisThriftCodec DATE_TIME_CODEC = new JodaDateTimeToEpochMillisThriftCodec(COMMON_CATALOG);
+    private static final ThriftCodecManager COMPILER_READ_CODEC_MANAGER = new ThriftCodecManager(new CompilerThriftCodecFactory(false), COMMON_CATALOG, ImmutableSet.of(DATA_SIZE_CODEC, DATE_TIME_CODEC));
```
Comment (ajaygeorge) [inline presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java:60]: ImmutableSet.of(DATA_SIZE_CODEC, DATE_TIME_CODEC) can be a constant var of its own than repeating it four times.

File: presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                FAKE_SUB_RESOURCE_GROUP_ID_2,
+                null,
+                null);
+        ResourceGroupInfo subResourceGroupInfo3 = getResourceGroupInfo(
+                FAKE_SUB_RESOURCE_GROUP_ID_3,
+                null,
+                null);
+        subGroups.add(subResourceGroupInfo1);
+        subGroups.add(subResourceGroupInfo2);
+        subGroups.add(subResourceGroupInfo3);
```
Comment (ajaygeorge) [inline presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java:139]: move this to a method called createSubGroups

**Review by ajaygeorge — APPROVED** [review]
LGTM % nits

**Comment by ajaygeorge** [comment]
@tdcmeehan Can you please take a look when you get a chance.

File: presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java (hunk 2 above)
Comment (Weizitong) [reply presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java:60]: Resolved locally. Will update in the next iter

File: presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java (hunk 3 above)
Comment (Weizitong) [reply presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java:139]: Resolved locally. Will update in the next iter

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**